community/baidu-search-plugin-starter/src/main/java/com/alibaba/cloud/plugin/BaiduSearchPluginConfiguration.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.cloud.plugin;
+
+import jdk.jfr.Description;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import java.util.function.Function;
+
+/**
+ * .
+ *
+ * @author: KrakenZJC
+ * @since : 2024-11-18
+ **/
+
+@Configuration
+@ConditionalOnClass(BaiduSearchService.class)
+public class BaiduSearchPluginConfiguration {
+
+	@Bean
+	@ConditionalOnMissingBean
+	@Description("Use baidu search engine to query for the latest news.") // function
+																			// description
+	public BaiduSearchService baiduSearchService() {
+		return new BaiduSearchService();
```

Comment:
Is it necessary to give users a method to set the behavior of BaiduSearch plugin? For example, by defining a  `BaiduSearchProperties` bean. 

It depends on the design of BaiduSearch. If it requires a token/key from user, then we must provide a way for users to set it. If it has some widely used parameters that will affect how it runs and the final result, we'd better make them configurable.

Reply:
```java
public BaiduSearchService baiduSearchService(BaiduSearchProperties properties) {
		return new BaiduSearchService(properties);
}
```